src/plone/restapi/testing.py

```diff
@@ -253,12 +253,16 @@ class RelativeSession(requests.Session):
     base if their URL is relative (doesn't begin with a HTTP[S] scheme).
     """
 
-    def __init__(self, base_url):
+    def __init__(self, base_url, test=None):
```

Comment:
@rpatterson can we document the "test" param here? Why we need it, what kind of values we expect, etc. I'd like to be a bit more expressive here, so that other devs in the future have a chance to understand why we did this here.

Reply:
This parameter shouldn't be required, not to have more warnings in the future?

Reply:
I don't quite follow, @wesleybl. It is optional, isn't it? Can you elaborate on your question?

Reply:
> @rpatterson can we document the "test" param here?

Good catch and apologies for not documenting in the code originally.  Done.

Reply:
> I don't quite follow, @wesleybl. It is optional, isn't it? Can you elaborate on your question?

I'm questioning if the test parameter shouldn't be required. If a new test is created and the test parameter is not passed, we will have new warnings.

Reply:
> > I don't quite follow, @wesleybl. It is optional, isn't it? Can you elaborate on your question?
> 
> I'm questioning if the test parameter shouldn't be required. If a new test is created and the test parameter is not passed, we will have new warnings.

Ah, you're suggesting maybe we should make it required.  Thanks for clarifying.  I made it optional because I figured making it opt-in would be less disruptive to other packages that use `RelativeSession`.  I know of at least one external usage and I didn't want to require that everything be released at once.

Reply:
Understand. The ideal then is to fix these external uses, after a release, and come back here to make it required.

Reply:
> Understand. The ideal then is to fix these external uses, after a release, and come back here to make it required.

I'm not sure I agree with making it required since I don't  think it even belongs there.  See my write up in the commit message.

Reply:
>I'm not sure I agree with making it required since I don't think it even belongs there

For test classes with `self.api_session` attribute it would be possible to make in a "Base Test Class". But I understood in your comment that it can be done at another time. Thanks!

Reply:
> For test classes with `self.api_session` attribute it would be possible to make in a "Base Test Class".

Yup, that's exactly what I imagined if I refactored the tests for DRY.